extensions/sql-migration/src/scripts/SHIR-auto-configuration.ps1

```diff
@@ -0,0 +1,609 @@
+	<#
+    .DESCRIPTION
+    Checks if the user executing this script has admin access.
+
+    .OUTPUTS
+    [bool]: true if user executing the script has admin access else false.
+    #>
+	$currentUser = New-Object Security.Principal.WindowsPrincipal $([Security.Principal.WindowsIdentity]::GetCurrent())
+	return $currentUser.IsInRole([Security.Principal.WindowsBuiltinRole]::Administrator)
+}
+
+#################################### Install an MSI package ###################################
+Function Install-MsiPackage {
+	<#
+    .DESCRIPTION
+    Installs the msi package and then verifies the installation.
+
+    .OUTPUTS
+    [bool]: True if installation was successful, else false
+    #>
+	Param(
+		# Name of the package to be installed
+		[Parameter(Mandatory)]
+		[string] $packageName,
+		# Full path of the package along with executable name
+		[Parameter(Mandatory)]
+		[string] $packagePath,
+		# Function that checks if package is installed
+		[Parameter(Mandatory)]
+		[scriptblock] $installationVerificationCallback
+	)
+
+	try {
+		# Check if the installation is already done
+		Write-OutputAndLog "Checking existing $packageName package installation..."
+		try {
+			$isPackageInstalled = & $installationVerificationCallback -ErrorAction SilentlyContinue
+			if ($null -ne $isPackageInstalled -and $isPackageInstalled -eq $true) {
+				Write-OutputAndLog "$packageName package is already installed."
+				return $true
+			}
+			else {
+				Write-OutputAndLog "$packageName package is not installed."
+			}
+		}
+		catch {
+			Write-ErrorAndLog -exception $_.Exception
+			return $false
+		}
+
+		# add necessary arguments to install quietly with no UI
+		# for no UI /qn, basic UI /qb, reduced UI /qr, full UI /qf
+		$argsx = @('/i', $packagePath, '/quiet', '/qb')
+
+		# execute installation process
+		Write-OutputAndLog "Starting $packageName package installation..."
+		$process = Start-Process msiexec -Wait -NoNewWindow -ErrorAction SilentlyContinue -ArgumentList $argsx
+		Write-OutputAndLog ("MSI Installation process exited with code: " + $process.ExitCode)
+		# after installation update the environment variable for the powershell session
+		$env:Path = [System.Environment]::GetEnvironmentVariable("Path", [System.EnvironmentVariableTarget]::Machine)
+
+		# Verifying the installation
+		Write-OutputAndLog "Verifying $packageName package installation..."
+		try {
+			$installationResult = & $installationVerificationCallback -ErrorAction SilentlyContinue
+			if ($null -ne $installationResult -and $installationResult -eq $true) {
+				Write-OutputAndLog "$packageName package installed sucessfully."
+				return $true
+			}
+			else {
+				Write-OutputAndLog "Failed to verify $packageName installation."
+				return $false
+			}
+		}
+		catch {
+			Write-ErrorAndLog -exception $_.Exception
+			# Only throw the error incase there is a valid verification command
+			if (-not [string]::IsNullOrEmpty($installationVerificationCallback.ToString())) {
```

Comment:
Would IsNullOrWhiteSpace be better suited here?

Reply:
Switched to isNullOrWhiteSpace!